libnmstate/netinfo.py

```diff
@@ -45,17 +46,36 @@ def interfaces():
 def devices():
     client = nmclient.client()
 
-    devs = client.get_devices()
+    devlist = []
+    for device in client.get_devices():
+        ip4info = {}
```

Comment:
same here, please try to split IP logic and the rest, split it into more functions